src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNetworkTask.java

```diff
@@ -158,13 +160,17 @@ private void insertItem(Block inputTarget, int previousSlot, ItemStack item) {
     private ItemStack distributeItem(ItemStack stack, Location inputNode, List<Location> outputNodes) {
         ItemStack item = stack;
 
-        Deque<Location> destinations = new LinkedList<>(outputNodes);
         Config cfg = BlockStorage.getLocationInfo(inputNode);
         boolean roundrobin = Objects.equals(cfg.getString("round-robin"), "true");
         boolean smartFill = Objects.equals(cfg.getString("smart-fill"), "true");
 
+        Collection<Location> destinations;
         if (roundrobin) {
-            roundRobinSort(inputNode, destinations);
+            Deque<Location> tempDestinations = new LinkedList<>(outputNodes);
+            roundRobinSort(inputNode, tempDestinations);
+            destinations = tempDestinations;
+        } else {
+            destinations = new ArrayList<>(outputNodes);
         }
```

Comment:
Some comments here would be nice.

Reply:
I just noticed an ArrayDeque might be faster; I'll redo the test and add comments

Reply:
ooh, nice

Reply:
it was indeed faster!